src/RunTests/RunVSTestTask.cs

```diff
@@ -0,0 +1,443 @@
+
+        /// <summary>
+        /// Gets or Sets Time out for hang source blame.
+        /// </summary>
+        public string VSTestBlameHangTimeout { get; set; }
+
+        /// <summary>
+        /// Gets or Sets The directory path where trace data collector is.
+        /// </summary>
+        public string VSTestTraceDataCollectorDirectoryPath { get; set; }
+
+        /// <summary>
+        /// Gets or Sets disabling Microsoft logo while running test through VSTest.
+        /// </summary>
+        public string VSTestNoLogo { get; set; }
+
+        /// <summary>
+        /// Gets or Sets Test artifacts processing mode which is applicable for .NET 5.0 or later versions.
+        /// </summary>
+        public string VSTestArtifactsProcessingMode { get; set; }
+
+        /// <summary>
+        /// Gets or Sets Correlation Id of test session.
+        /// </summary>
+        public string VSTestSessionCorrelationId { get; set; }
+
+        /// <summary>
+        /// Gets or Sets Runner version of VSTest.
+        /// </summary>
+        public string VSTestRunnerVersion { get; set; }
+
+        /// <summary>
+        /// Gets or Sets Path to nuget package cache.
+        /// </summary>
+        public string NugetPath { get; set; }
+
+        /// <summary>
+        /// Executes the test. Skips execution if specified.
+        /// </summary>
+        /// <returns>Returns true if the test was executed, otherwise false.</returns>
+        public override bool Execute()
+        {
+            var traceEnabledValue = Environment.GetEnvironmentVariable("VSTEST_BUILD_TRACE");
+            if (SkipExecution)
+            {
+                if (!string.IsNullOrEmpty(traceEnabledValue) && traceEnabledValue.Equals("1", StringComparison.Ordinal))
+                {
+                    Log.LogMessage("Skipping test execution.");
+                }
+
+                return true;
+            }
+
+            var debugEnabled = Environment.GetEnvironmentVariable("VSTEST_BUILD_DEBUG");
+            if (!string.IsNullOrEmpty(debugEnabled) && debugEnabled.Equals("1", StringComparison.Ordinal))
+            {
+                Log.LogMessage("Waiting for debugger attach...");
+
+                var currentProcess = Process.GetCurrentProcess();
+                Log.LogMessage($"Process Id: {currentProcess.Id}, Name: {currentProcess.ProcessName}");
+
+                while (!Debugger.IsAttached)
+                {
+                    Thread.Sleep(1000);
+                }
+
+                Debugger.Break();
+            }
+
+            // Avoid logging "Task returned false but did not log an error." on test failure, because we don't
+            // write MSBuild error. https://github.com/dotnet/msbuild/blob/51a1071f8871e0c93afbaf1b2ac2c9e59c7b6491/src/Framework/IBuildEngine7.cs#L12
+            var allowfailureWithoutError = BuildEngine.GetType().GetProperty("AllowFailureWithoutError");
+            allowfailureWithoutError?.SetValue(BuildEngine, true);
```

Comment:
This is a bit of a hack. Should an error just be logged to give the user some indication of what's happening?